client/src/components/Doctor/UpdateProfile.tsx

```diff
@@ -18,23 +18,16 @@ const UpdateProfile: FunctionComponent<{}> = () => {
   const [address, setAddress] = useState('');
   const [hc, setHc] = useState('');
   const { user } = useContext(AuthContext);
-  const { updateProfile, currentAccount, raiseDoctorToHCRequest } = useContext(HealthContext);
+  const { updateProfile, currentAccount, raiseDoctorToHCRequest, fetchHCProviders, hcProviderArr } =
+    useContext(HealthContext);
+
   const account = currentAccount || '';
 
-  let providers = [
-    {
-      name: 'Kaiser',
-      address: '0xB74357Df49Ed35Ec1AEc5efdA41f5A5D846fAa8e',
-    },
-    {
-      name: 'Sutter Health Care',
-      address: '0xB3CC507e752Dcc3DA1cEf955B58e97Ae77160103',
-    },
-    {
-      name: 'El Camino',
-      address: '0x8eda1014b9177d464306935e8fcf9fd27c20aa08',
-    },
-  ];
+  useEffect(() => {
+    (async () => {
+      await fetchHCProviders?.();
```

Comment:
please fetch hcproviders and store in component, do not use global state for blockchain and api call results. only profile information of user is supposed to be stored in global state.